src/Game/Connect4/Game.elm

```diff
@@ -0,0 +1,170 @@
+module Game.Connect4.Game exposing (game)
+
+import Dict exposing (Dict)
+import Game exposing (GameMsg(..), InboundMsg(..), PlayerId)
+import Game.Connect4.Engine exposing (engine)
+import Game.Connect4.Helpers exposing (cols)
+import Game.Connect4.Types as Types exposing (EngineMsg(..), Model, Player(..), State, Tile(..), ViewMsg(..))
+import Html exposing (Html, div, text)
+import Html.Attributes exposing (class)
+import Html.Events exposing (onClick)
+import List.Extra as ListEx
+import Time
+
+
+game : Game.Game Model State EngineMsg ViewMsg
+game =
+    { view = view
+    , css = css
+    , init = init
+    , update = update
+    , subscriptions = subscriptions
+    , gameMsgEncoder = Types.encodeViewMsg
+    , gameMsgDecoder = Types.decodeViewMsg
+    , modelEncoder = Types.encodeModel
+    , modelDecoder = Types.decodeModel
+    , engine = engine
+    }
+
+
+init : PlayerId -> State -> ( Model, Maybe (GameMsg EngineMsg ViewMsg), Cmd (GameMsg EngineMsg ViewMsg) )
+init playerId state =
+    let
+        msg =
+            if Dict.size state.players < 2 then
```

Comment:
FYI this is the auto-join code. Technically you could you add a "join as 🔴" button or whatever to give users a choice to join.

Reply:
I kinda like how easy it is that it's just a server message, nothing too special about joining a game.